dd-sdk-android-gradle-plugin/src/test/kotlin/com/datadog/gradle/plugin/DdAndroidGradlePluginFunctionalTest.kt

```diff
@@ -0,0 +1,323 @@
+            .build()
+
+        // Then
+        assertThat(result.task(":$taskName")?.outcome)
+            .isEqualTo(TaskOutcome.SUCCESS)
+    }
+
+    @Test
+    fun `M not contain any uploadTasks W minifyNotEnabeld`() {
+        // Given
+        stubGradleBuildFromResourceFile("build_with_minify_not_enabled.gradle.kts")
+
+        // When
+        val result = GradleRunner.create()
+            .withProjectDir(testProjectDir)
+            .withArguments("tasks")
+            .withPluginClasspath(getTestConfigurationClasspath())
+            .build()
+
+        // Then
+        val uploadTask = result.output
+            .split("\n")
+            .firstOrNull { it.startsWith("upload") }
+        assertThat(uploadTask).isNull()
+    }
+
+    // endregion
+
+    // region Internal
+
+    private fun stubFile(destination: File, content: String) {
+        with(destination.outputStream()) {
+            write(content.toByteArray())
+        }
+    }
+
+    private fun stubGradleBuildFromResourceFile(resourceFilePath: String) {
+        javaClass.classLoader.getResource(resourceFilePath)?.file?.let {
+            File(it).copyTo(buildFile)
+        }
+    }
+
+    private fun getTestConfigurationClasspath(): List<File> {
+        // we will use this classpath for the GradleRunner to make sure we have all the
+        // required classes (AndroidGradle and KotlinGradle) to build the plugin. GradleRunner
+        // creates its own process and by default the classpath used is the one from the `main`
+        // configuration.
+        val properties = Properties()
+        properties["implementation-classpath"] = System.getProperty("java.class.path")
+        return PluginUnderTestMetadataReading
+            .readImplementationClasspath("gradle-runner-classpath", properties)
+    }
+
+    // endregion
+
+    companion object {
+        val ANDROID_MANIFEST_FILE_CONTENT = """
+            <?xml version="1.0" encoding="utf-8"?>
+            <manifest xmlns:android="http://schemas.android.com/apk/res/android"
+                      package="com.example.variants">
+
+                <application
+                    android:allowBackup="true"
+                    android:supportsRtl="true">
+                    <activity android:name=".MainActivity">
+                        <intent-filter>
+                            <action android:name="android.intent.action.MAIN"/>
+
+                            <category android:name="android.intent.category.LAUNCHER"/>
+                        </intent-filter>
+                    </activity>
+                </application>
+
+            </manifest>
+        """.trimIndent()
+
+        const val SETTINGS_FILE_CONTENT = """
+            pluginManagement {
+                resolutionStrategy {
+                    eachPlugin {
+                        if (requested.id.id == "com.android.application") {
+                            useModule("com.android.tools.build:gradle:7.0.3")
```

Comment:
ideally versions here should be taken from versions catalog and be the same as we use in the plugin dependencies, but tbh I don't know how to reference them.

Maybe some property which we can inject from `buildSrc` into test environment? Same for the DD SDK version in the test build script files.

Reply:
Yeah, this is going to be tricky I guess. I am thinking that maybe we can do this in another task as there are already several other things to tackle into this one.